### DISCUSSION TITLE
Understanding error messages

### DISCUSSION THREAD

**Discussion by ajtruckle** [MicrosoftEdge/WebView2Feedback#4344] [Q&A]
@champnic  What do these error messages mean on Windows computers?

* Cannot create webview environment. (0x80070002).
* Cannot create webview environment. (0x800703E6).

My application is Win32 using the runtime.

**Comment by victorhuangwq** [comment]
@lflores-ms could you help @ajtruckle out?

**Reply by ajtruckle** [reply]
@victorhuangwq @lflores-ms Is there no chance of an explanation?

**Comment by lflores-ms** [comment]
Hi @ajtruckle -- Please find error description for environment creation in the [Globals section](https://learn.microsoft.com/en-us/microsoft-edge/webview2/reference/win32/webview2-idl?view=webview2-1.0.2277.86#createcorewebview2environment:~:text=Expand%20table-,Error%20value,-Description) of our docs.

* 0x80070002 is the same as `HRESULT_FROM_WIN32(ERROR_FILE_NOT_FOUND)`. Please make sure that the WebView2 Runtime is installed in the device. You can check for the runtime with [GetAvailableCoreWebView2BrowserVersionString](https://learn.microsoft.com/en-us/microsoft-edge/webview2/reference/win32/webview2-idl?view=webview2-1.0.2357-prerelease#getavailablecorewebview2browserversionstring) API.

* 0x800703E6 is `ERROR_NOACCESS`. This might be due to no access to [User Data Folder](https://learn.microsoft.com/en-us/microsoft-edge/webview2/concepts/user-data-folder?tabs=win32#suitable-custom-udf-locations). Please make sure the UDF is in a location accessible by the app.

Are you observing any of these issues with a specific version of the WebView2 Runtime or SDK?
Debugger output might have more details if you still hit these issues.

**Reply by ajtruckle** [reply]
@lflores-ms Thanks for these explanations. One user was able to resolve the error by upgrading the evergreen runtime.

**Reply by ajtruckle** [reply]
@lflores-ms The linked article does not mention anywhere the 0x... wording. So we can't search and locate these terms.

Since the article has the ID versions listed then maybe the errors should use those IDs too?

It some be much easier for anyone to find out what a specific error message means in the online resources.
